### PR TITLE
feat: add koordinates read role TDE-946

### PR DESCRIPTION
**Pull request by MDavidson17**
### Motivation

In order for data in nz-imagery to be used as the publishing bucket, koordinates needs read role access to the ODR buckets.

### Modification
KxReadRole create in stack and applied to ODR dataset buckets

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1021,13 +1021,11 @@
     },
     "node_modules/aws-cdk-lib/node_modules/@balena/dockerignore": {
       "version": "1.0.2",
```
Comment (blacha) [inline package-lock.json]: Might be a idea to revert these changes?

**Review by l0b0 — APPROVED** [review]

**Review by blacha — APPROVED** [review]
